compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -792,6 +792,10 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     );
                 }
             }
+            ObligationCauseCode::LetElse => {
+                err.help("try adding a `return` or `panic!(..)`");
+                err.help("...or use `match` instead of `let...else`");
```

Comment:
```suggestion
                err.help("...or use either `match` or `if...let` instead of `let...else`");
```

I'd guess that in a single match non-diverging case, `if-let` is probably what the person wants.

Reply:
That is possible. I think this is a "less is more" scenario. I decided on just `match` here because `if let` does not allow you to create bindings in scope for later statements. For a beginner, one suggestion that will always work is more helpful IMO.